test-framework/keycloak-server/src/main/java/io/quarkus/test/keycloak/client/KeycloakTestClient.java

```diff
@@ -24,10 +24,17 @@
 public class KeycloakTestClient implements DevServicesContext.ContextAware {
 
     private final static String CLIENT_AUTH_SERVER_URL_PROP = "client.quarkus.oidc.auth-server-url";
+
```

Comment:
I think `KeycloakTestClient` should remain unchanged,  it is about helping tests to acquire tokens from OIDC server, the OIDC client does not need help from `KeycloakTestClient`. 
Well, may be only `KeycloakTestClient` might help to upload realms, and it is only  `OIDC_CLIENT_AUTH_SERVER_URL_CONFIG_KEY` that would be relevant.

But as far as the non admin token acquisition goes, I'm worried we may get some strange errors when both OIDC server and OIDC client are loaded, `quarkus-oidc` tests for acquiring tokens from OIDC server will pick oidc-client properties pointing to different realms, etc.

As far as this PR is concerned I propose not to deal with KeycloakTestClient updates. We can consider something later

Reply:
It is not a problem, changes in the `KeycloakTestClient` are not required and I'll remove them.

If you don't mind, I think you very much misunderstood changes here, I am simply adding support to use `KeycloakTestClient` when OIDC extension is not present. I understand it won't be frequent as this is OIDC client, but I still can imagine users will want to get token in the test directly.

Anyway, it was just an idea, I'll remove it. Thanks for nice explanation.

Reply:
Hi @michalvavrik, how one would use it though if the Quarkus endpoint is not expected to be protected by quarkus-oidc, that is the question. But yeah, we can iterate on it earlier once users tell us a bit more about their OIDC client plus KC devservice experience